projects/wpdx/src/app/rehab-prio/adm-popup/adm-popup.component.ts

```diff
@@ -71,80 +71,53 @@ export class AdmPopupComponent implements OnChanges {
     wpdx_plus`
     const queries: string[] = [];
-    if (value.NAME_0) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' group by ${attribute}`);
-    }
-    if (value.NAME_1) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' group by ${attribute}`);
+    if (value.NAME_0){
+      baseQuery = `${baseQuery} where clean_country_name='${this.fq(value.NAME_0)}'`;
+      queries.push(`${baseQuery} group by ${attribute}`);
     }
-    if (value.NAME_2) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' group by ${attribute}`);
-    }
-    if (value.NAME_3) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' group by ${attribute}`);
-    }
-
-    if (value.NAME_4) {
-      queries.push(`${baseQuery}
-        where clean_country_name='${this.fq(value.NAME_0)}' and clean_adm1='${this.fq(value.NAME_1)}' and 
-              clean_adm2='${this.fq(value.NAME_2)}' and clean_adm3='${this.fq(value.NAME_3)}' and clean_adm4='${this.fq(value.NAME_4)}' group by ${attribute}`);
+    for (let i = 0; i < values.length; i++){
+      baseQuery = this.addQuery(values[i], baseQuery, i+1, queries, attribute)
     }
     return queries;
 
   }
 
-  getManagementData(){
-    forkJoin(this.getDataQuery(this.popupProperties, 'management_clean').map(q => this.db.query(q))).subscribe(results => {
-      this.management_data = [];
+  getData(column_name){
+    forkJoin(this.getDataQuery(this.popupProperties, column_name).map(q => this.db.query(q))).subscribe(results => {
+      const data_array = [];
       for (let i = 0; i < results.length; i++){
         const x = [];
-        for (let j = 0; j < results[i].rows.length; j++){
+        for (let j = 0; j < results[i].rows.length; j ++){
           if (results[i].rows[j].count > 0){
-            x.push({'name': results[i].rows[j].management_clean, 'value': results[i].rows[j].count});
-          }
-      }
-        this.management_data.push(x);
-      }
-    })
-  }
-
-  getTechData(){
-    forkJoin(this.getDataQuery(this.popupProperties, 'water_tech_clean').map(q => this.db.query(q))).subscribe(results => {
-      this.tech_data = [];
-      for (let i = 0; i < results.length; i++){
-        const x = [];
-        for (let j = 0; j < results[i].rows.length; j++){
-          if (results[i].rows[j].count > 0){
-            x.push({'name': results[i].rows[j].water_tech_clean, 'value': results[i].rows[j].count});
+            x.push({'name': Object.values(results[i].rows[j])[1], 'value': results[i].rows[j].count});
```

Comment:
There's no guarantee for the order of items in `Object.values()`, so you can't rely on anything being in index 1.

Reply:
If you need the 'name' column name here, you can either fix it in the query itself (`select column as something...`) or make `getDataQuery` return a list of `Observable`s and not strings.